src/Tasks/Microsoft.NET.Build.Tasks/CacheArtifactParser.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+using NuGet.Versioning;
+using NuGet.Packaging.Core;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    //The following provides the logic to parse the output - artifact.xml
+    // generated by ComposeCache
+    internal class CacheArtifactParser
+    {
+        internal static HashSet<PackageIdentity> Parse(string filterProjFilesList)
+        {
+            var listofFilterFiles = filterProjFilesList.Split(';');
+            var listofPackages = new HashSet<PackageIdentity>();
+
+            foreach (var filterFile in listofFilterFiles)
+            {
+                var doc = XDocument.Load(filterFile);
+                var ns = doc.Root.Name.Namespace;
+
+                foreach (var pkginfo in doc.Root.Elements(ns + "ItemGroup").Elements("PackageReference"))
+                {
+                    var pkgname = pkginfo.Attribute("Include");
+                    var version = pkginfo.Attribute("Version");
```

Comment:
If you insist on using MSBuild as the format of this file, then you should respect all ways that MSBuild can express Items.  This code doesn't handle the format:

```
<PackageReference Include="Newtonsoft.Json">
  <Version>9.0.1</Version>
</PackageReference>
```